### PR TITLE
[10.0][IMP] shopinvader: Allows to display not bound products in cart

### PR DESCRIPTION
**Pull request by rousseldenis**
In some flows, people should be able to modify cart from backend.
e.g.: To propose to customer a better alternative to selected product or...

### CODE REVIEW

**Comment by rousseldenis** [comment]
@Cedric-Pigeon 
@lmignon

File: shopinvader/models/shopinvader_backend.py [hunk 1]
```diff
@@ -100,6 +100,11 @@ class ShopinvaderBackend(models.Model):
         "statistics reasons. A new cart is created automatically when the "
         "customer will add a new item.",
     )
+    authorize_not_binded_products = fields.Boolean(
```
Comment (simahawk) [inline shopinvader/models/shopinvader_backend.py]: please, let's start using proper wording: "bound" instead of "binded" :pray: :angel:

Comment (Cedric-Pigeon) [inline shopinvader/models/shopinvader_backend.py]: adding it on a view would be useful....

**Review by Cedric-Pigeon — CHANGES_REQUESTED** [review]

**Comment by Cedric-Pigeon** [comment]
@rousseldenis Please check Travis.

**Review by simahawk — CHANGES_REQUESTED** [review]

File: shopinvader/services/abstract_sale.py [hunk 2]
```diff
@@ -43,14 +46,28 @@ def _convert_step(self, sale):
...
         return True
 
-    def _convert_one_line(self, line):
+    def _get_product_to_convert(self, line):
+        """
+        Get the product to display on cart line
+        :param line:
+        :return: product.product
+        """
         if line.shopinvader_variant_id:
```
Comment (simahawk) [inline shopinvader/services/abstract_sale.py:55]: the computed field is buggy. You could cherry pick this fix https://github.com/shopinvader/odoo-shopinvader/pull/528/commits/617032d72b0512cf5cadecf522f3b8be1d6b9996

Comment (rousseldenis) [reply shopinvader/services/abstract_sale.py:55]: Your fix would work only if you don't use guest mode.

Comment (rousseldenis) [reply shopinvader/services/abstract_sale.py:55]: I think you should be interested by this one : https://github.com/shopinvader/odoo-shopinvader/pull/341

**Comment by rousseldenis** [comment]
@simahawk

File: shopinvader/services/abstract_sale.py (hunk 2 above)
Comment (simahawk) [reply shopinvader/services/abstract_sale.py:55]: @rousseldenis tnx for the reference! 
I have no time to check it in details but it seems that we should have a mix of both (my fix and the one from #341).
What do you think?

**Comment by Cedric-Pigeon** [comment]
@rousseldenis Can you please finish this one?

File: shopinvader/services/abstract_sale.py (hunk 2 above)
Comment (rousseldenis) [reply shopinvader/services/abstract_sale.py:55]: @simahawk Could we merge this and then fix the field ? I can then make a PR on your repo with these modifications adapted.

Comment (rousseldenis) [reply shopinvader/services/abstract_sale.py:55]: @sebastienbeau Waht do you think ?

**Review by simahawk — APPROVED** [review]
unlocking it :)

**Comment by rousseldenis** [comment]
@simahawk Thanks. Changed also the typo in commit message.
@Cedric-Pigeon

**Review by lmignon — APPROVED** [review]
LGTM (code review)

**Review by sebastienbeau — APPROVED** [review]
LGTM code review
/ocabot merge minor

**Pull request merged**